test/groovy/ContainerExecuteStructureTestsTest.groovy

```diff
@@ -0,0 +1,154 @@
+        })
+        jsr.step.containerExecuteStructureTests(
+            script: nullScript,
+            juStabUtils: utils,
+            containerCommand: '/busybox/tail -f /dev/null',
+            containerShell: '/busybox/sh',
+            dockerImage: 'myRegistry:55555/pathTo/myImage:myTag',
+            testConfiguration: 'cst/*.yml',
+            testImage: 'myRegistry/myImage:myTag'
+        )
+        // asserts
+        assertThat(jscr.shell, hasItem(allOf(
+            stringContainsInOrder(['#!/busybox/sh', 'container-structure-test', '--config']),
+            containsString('--config cst\\test1.yml'),
+            containsString('--config cst\\test2.yml'),
+            containsString('--driver docker'),
+            containsString('--image myRegistry/myImage:myTag'),
+            containsString('--test-report ./cst-report.json'),
+        )))
+        //currently no default Docker image
+        assertThat(jedr.dockerParams.dockerImage, is('myRegistry:55555/pathTo/myImage:myTag'))
+        assertThat(jedr.dockerParams.dockerOptions, is("-u 0 --entrypoint=''"))
+        assertThat(jedr.dockerParams.containerCommand, is('/busybox/tail -f /dev/null'))
+        assertThat(jedr.dockerParams.containerShell, is('/busybox/sh'))
+        assertThat(jlr.log, containsString('{testResult: true}'))
+        assertThat(jscr.shell, hasItem('docker pull myRegistry/myImage:myTag'))
+    }
+
+    @Test
+    void testExecuteContainterStructureTestsK8S() throws Exception {
+        def envDefault = nullScript.env
+        nullScript.env = [ON_K8S: 'true']
+        jsr.step.containerExecuteStructureTests(
+            script: nullScript,
+            juStabUtils: utils,
+            dockerImage: 'myRegistry:55555/pathTo/myImage:myTag',
+            testConfiguration: 'cst/*.yml',
+            testImage: 'myRegistry/myImage:myTag'
+        )
+        nullScript.env = envDefault
+        // asserts
+        assertThat(jscr.shell, hasItem(allOf(
+            stringContainsInOrder(['#!/bin/sh', 'container-structure-test', '--config']),
+            containsString('--config cst\\test1.yml'),
+            containsString('--config cst\\test2.yml'),
+            containsString('--driver tar'),
+            containsString('--image myRegistry/myImage:myTag'),
+            containsString('--test-report ./cst-report.json'),
+        )))
+        assertThat(jedr.dockerParams.dockerImage, is('myRegistry:55555/pathTo/myImage:myTag'))
+        assertThat(jscr.shell, not(hasItem('docker pull myRegistry/myImage:myTag')))
+    }
+
+    @Test
+    void testExecuteContainterStructureTestsError() throws Exception {
+        helper.registerAllowedMethod('readFile', [String.class], {s ->
+            return '{testResult: true}'
+        })
+        helper.registerAllowedMethod('sh', [String.class], {s ->
+            if (s.startsWith('#!/busybox/sh\ncontainer-structure-test test')) {
+                throw new GroovyRuntimeException('shell call failed')
+            } else {
```

Comment:
you could ommit putting it into an  `else` since the `throw` anyways breaks the flow.